usermanagerlib/src/test/java/org/pesmypetcare/usermanagerlib/clients/WeightManagerClientTest.java

```diff
@@ -0,0 +1,143 @@
+        weight = new Weight(weightData);
+        weightList = new ArrayList<>();
+        weightList.add(weight);
+        jsonWeightData = new StringBuilder("{\n"
+            + "      \"weight\": 5.4\n"
+            + "    }\n");
+        jsonAllWeights = new StringBuilder("[\n"
+            + "{\n"
+            + "    \"date\": \"2020-02-13T10:30:00\",\n"
+            + "    \"body\": {\n"
+            + "      \"weight\": 5.4\n"
+            + "    }\n"
+            + "  }"
+            + "   \n"
+            + "]");
+    }
+
+    @Test
+    public void createWeight() throws ExecutionException, InterruptedException {
+        given(taskManager.resetTaskManager()).willReturn(taskManager);
+        given(taskManager.execute(anyString(), anyString())).willReturn(taskManager);
+        given(taskManager.get()).willReturn(STATUS_OK);
+
+        int responseCode = client.createWeight(ACCESS_TOKEN, owner, petName, weight, date);
+        assertEquals("Should return response code 200", STATUS_OK_INT, responseCode);
+    }
+
+    @Test
+    public void deleteByDate() throws ExecutionException, InterruptedException {
+
+        given(taskManager.resetTaskManager()).willReturn(taskManager);
+        given(taskManager.execute(anyString(), anyString())).willReturn(taskManager);
+        given(taskManager.get()).willReturn(STATUS_OK);
+
+        int responseCode = client.deleteByDate(ACCESS_TOKEN, owner, petName, date);
+        assertEquals("Should return response code 200", STATUS_OK_INT, responseCode);
+    }
+
+    @Test
+    public void deleteAllWeights() throws ExecutionException, InterruptedException {
+        given(taskManager.resetTaskManager()).willReturn(taskManager);
+        given(taskManager.execute(anyString(), anyString())).willReturn(taskManager);
+        given(taskManager.get()).willReturn(STATUS_OK);
+
+        int responseCode = client.deleteAllWeight(ACCESS_TOKEN, owner, petName);
+        assertEquals("Should return response code 200", STATUS_OK_INT, responseCode);
+    }
+
+    @Test
+    public void getWeightData() throws ExecutionException, InterruptedException {
+        given(taskManager.resetTaskManager()).willReturn(taskManager);
+        given(taskManager.execute(anyString(), anyString())).willReturn(taskManager);
+        given(taskManager.get()).willReturn(jsonWeightData);
+
+        WeightData response = client.getWeightData(ACCESS_TOKEN, owner, petName, date);
+        assertEquals("Should return meal data", response, weightData);
+    }
+
+    @Test
+    public void getAllMealData() throws ExecutionException, InterruptedException {
+        given(taskManager.resetTaskManager()).willReturn(taskManager);
+        given(taskManager.execute(anyString(), anyString())).willReturn(taskManager);
+        given(taskManager.get()).willReturn(jsonAllWeights);
+
+        List<Weight> response = client.getAllWeightData(ACCESS_TOKEN, owner, petName);
+        assertEquals("Should return a meal data list", response, weightList);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String "Should return a meal data list" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5361615)